packages/graphql/tests/integration/issues/1735.int.test.ts

```diff
@@ -0,0 +1,140 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { graphql, GraphQLSchema } from "graphql";
+import type { Driver } from "neo4j-driver";
+import { Neo4jGraphQL } from "../../../src";
+import { generateUniqueType } from "../../utils/graphql-types";
+import Neo4j from "../neo4j";
+
+describe("https://github.com/neo4j/graphql/issues/1735", () => {
+    const actorType = generateUniqueType("Actor");
+    const movieType = generateUniqueType("Movie");
+
+    let schema: GraphQLSchema;
+    let driver: Driver;
+    let neo4j: Neo4j;
+
+    beforeAll(async () => {
+        neo4j = new Neo4j();
+        driver = await neo4j.getDriver();
+        const typeDefs = `
+          interface MovieActorEdgeProperties @relationshipProperties {
+            isLead: Boolean
+          }
+
+          type ${movieType.name} {
+              dbId: ID! @id(global: true) @alias(property: "id")
+              title: String!
+              actors: [${actorType.name}!]! @relationship(type: "ACTED_IN", direction: IN, properties: "MovieActorEdgeProperties")
+              leadActorsCount: Int! @cypher(statement:"""
+                MATCH (this)<-[rel:ACTED_IN]-(a:${actorType.name})
+                WHERE rel.isLead = true
+                RETURN count(a)
+              """)
+          }
+
+          type ${actorType.name} {
+              dbId: ID! @id(global: true) @alias(property: "id")
+              name: String!
+              movies: [${movieType.name}!]! @relationship(type: "ACTED_IN", direction: OUT, properties: "MovieActorEdgeProperties")
+          }
+  `;
+        const neoGraphql = new Neo4jGraphQL({ typeDefs, driver });
+        schema = await neoGraphql.getSchema();
+    });
+
+    beforeEach(async () => {
+        const source = `
+        mutation($input: [${movieType.name}CreateInput!]!) {
+          ${movieType.operations.create}(input: $input) {
+            ${movieType.plural} {
+              id
+              leadActorsCount
+            }
+          }
+        }
+      `;
+        const input = [
+            {
+                title: "A River Runs Through It",
+                actors: {
+                    create: {
+                        edge: { isLead: false },
+                        node: { name: "River Phoenix" },
+                    },
+                },
+            },
+        ];
+        await graphql({
+            source,
+            schema,
+            contextValue: { driver },
```

Comment:
Thank you for the updates @litewarp!
One last thing then this is good to go from my side, here and on line 124 please use `contextValue: neo4j.getContextValues()` that the query is executed against the correct database. Thanks!